persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/StargateQueryHandler.java

```diff
@@ -238,7 +240,8 @@ protected void authorizeByToken(Map<String, ByteBuffer> customPayload, CQLStatem
             authenticationSubject,
             castStatement.keyspace(),
             castStatement.columnFamily(),
-            SourceAPI.CQL);
+            SourceAPI.CQL,
+            EMPTY_HEADERS);
```

Comment:
Same concern as in https://github.com/stargate/stargate/pull/581/files#r559749974

Reply:
For stargate query handlers we should pass `EMPTY_HEADERS`. Only the CNDB `StargasteQueryHandler` will extract the tenant id from `ClientStateWithTenant` and pass it to the components. It was discussed on slack.

Reply:
This should pass all the cql properties. This logic needs to be generic and apply to all backends.

Reply:
We were discussing this on the #stargate-team. We don't have access to headers nor `ClientState` here. The conclusion was that we need to extract it from the `ClientState` and pass further only for the CNDB's `StargateQueryHandler`

Reply:
Implemented in:
https://github.com/stargate/stargate/pull/581/commits/05d379881dcf18262d3d07eaf21ccb33dad3960c